front/components/SearchBar.tsx

```diff
@@ -0,0 +1,61 @@
+import { Input, Column, Text, Box, Button, Pressable } from "native-base";
+import React from "react";
+
+interface SearchBarProps {
+	onTextChange: (text: string) => void;
+	onTextSubmit: (text: string) => void;
+	suggestions: string[];
+}
+
+// debounce function
+const debounce = (func: any, delay: number) => {
+	let inDebounce: any;
+	return function (this: any) {
+		const context = this;
+		const args = arguments;
+		clearTimeout(inDebounce);
+		inDebounce = setTimeout(() => func.apply(context, args), delay);
+	};
+};
+
+const SearchBar = ({
+	onTextChange,
+	onTextSubmit,
+	suggestions,
+}: SearchBarProps) => {
+	const debouncedOnTextChange = React.useRef(
+		debounce((t: string) => onTextChange(t), 70)
+	).current;
+	return (
+		<>
```

Comment:
The input filed should not take the whole width of the viewport. It should be defined using a ratio, and with a maxWidth value